### PR TITLE
Dealing with the problem that tf.name_scope is not reflected #17538

### PR DESCRIPTION
**Pull request by DameNianch**
Fixed the problem that tf.name_scope is ignored even if tf.keras.__internal__.apply_name_scope_on_model_declaration is used.

related issues
- https://github.com/keras-team/keras/issues/17538
- https://github.com/tensorflow/tensorflow/issues/27298

PS. Thanks to @slanzmich for finding this solution.

### CODE REVIEW

**Comment by sachinprasadhs** [comment]
Hi, Thanks for the PR, please use the proper commit message to understand the Gist of the changes made. https://cbea.ms/git-commit/

**Comment by DameNianch** [comment]
> Hi, Thanks for the PR, please use the proper commit message to understand the Gist of the changes made. https://cbea.ms/git-commit/

Sincerest apologies. I have made the correction based on the link you provided. If there are still any issues, we will correct.

**Comment by rchao** [comment]
Thank you for the PR! 

@pradeepkuppla, would it be possible to give this a run across all internal tests before we proceed with an approval?

**Comment by DameNianch** [comment]
Is there anything I can do? I'm at a loss as to what I can do for the two checks.

**Comment by sachinprasadhs** [comment]
Hello, Thank you for submitting a pull request.

We're currently in the process of migrating the new `Keras 3 code base` from `keras-team/keras-core` to `keras-team/keras`. 
Consequently, merging this PR is not possible at the moment. After the migration is successfully completed, feel free to reopen this PR at `keras-team/keras` if you believe it remains relevant to the `Keras 3 code base`. If instead this PR fixes a bug or security issue in legacy `tf.keras`, you can instead reopen the PR at `keras-team/tf-keras`, which hosts the `TensorFlow-only`, legacy version of `Keras`.
